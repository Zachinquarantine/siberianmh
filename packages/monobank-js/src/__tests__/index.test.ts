import { Monobank } from '../'
import nock from 'nock'

nock.disableNetConnect()

describe('monobank', () => {
  test('getCurrency()', async () => {
    const mockResponse = [
      {
        currencyCodeA: 840,
        currencyCodeB: 980,
        date: 1601068206,
        rateBuy: 28.12,
        rateSell: 28.4083,
      },
      {
        currencyCodeA: 978,
        currencyCodeB: 980,
        date: 1601068206,
        rateBuy: 32.65,
        rateSell: 33.1104,
      },
    ]

    const mock = nock('https://api.monobank.ua')
      .get('/bank/currency')
      .reply(200, mockResponse)
    const mb = new Monobank()
    const result = await mb.getCurrency()

    expect(mock.isDone()).toBe(true)
    expect(result.data).toMatchSnapshot()
    expect(result.status).toBe(200)
  })
})
