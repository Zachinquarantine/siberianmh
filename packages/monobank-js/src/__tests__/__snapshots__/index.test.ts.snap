// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`monobank getCurrency() 1`] = `
Array [
  Object {
    "currencyCodeA": 840,
    "currencyCodeB": 980,
    "date": 1601068206,
    "rateBuy": 28.12,
    "rateSell": 28.4083,
  },
  Object {
    "currencyCodeA": 978,
    "currencyCodeB": 980,
    "date": 1601068206,
    "rateBuy": 32.65,
    "rateSell": 33.1104,
  },
]
`;
